src/axom/quest/detail/shaping/WindingNumberSampler.hpp

```diff
@@ -0,0 +1,326 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_QUEST_WINDING_NUMBER_SAMPLER__HPP_
+#define AXOM_QUEST_WINDING_NUMBER_SAMPLER__HPP_
+
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+#include "axom/primal.hpp"
+#include "axom/mint.hpp"
+#include "axom/spin.hpp"
+#include "axom/quest/InOutOctree.hpp"
+#include "axom/quest/detail/shaping/shaping_helpers.hpp"
+
+#include "axom/fmt.hpp"
+
+#include "mfem.hpp"
+
+namespace axom
+{
+namespace quest
+{
+namespace shaping
+{
+
+using QFunctionCollection = mfem::NamedFieldsMap<mfem::QuadratureFunction>;
+using DenseTensorCollection = mfem::NamedFieldsMap<mfem::DenseTensor>;
+
+namespace detail
+{
+
+/*!
+ * \brief Check whether point \a pt is inside \a shape.
```

Comment:
Please mention that this is using the non-zero rule to determine containment

Reply:
Done.